source/TimePicker.js

```diff
@@ -585,14 +602,32 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
+			var meridiems = event.originator.get('meridiems'),
+				oldMeridiem = meridiems[event.old],
+				newMeridiem = meridiems[this.$.meridiem.get('value')],
+				oldHour = this.$.hour.get('value'),
+				startHour = parseInt(newMeridiem['start'], 10),
+				offset = oldHour - parseInt(oldMeridiem['start'], 10);
 
-			if (this.value) {
-				// value is 0 for am, 1 for pm
-				// reset the hour to < 12 and then add 12 if it's pm
-				hour = hour%12 + value*12;
-				this.updateHours(hour);
+			if (meridiems.length == 2) {
+				this.updateValue(startHour + offset);
+			} else {
+				var oldMinute =  this.$.minute.get('value'),
+					endHour = parseInt(newMeridiem['end'], 10),
```

Comment:
Let's use dot notation (i.e. `newMeridiem.end`) where possible. This applies to the following two lines, as well.